### DISCUSSION TITLE
The problem of exporting the application to (apk) on the Mac system

### DISCUSSION THREAD

**Discussion by avcr2019** [airsdk/Adobe-Runtime-Support#2696] [General]
welcome . I wish you a good day
I have a strange problem and I don't know how to solve it
I was using the Windows system to export my game through (Adobe animate)
But I switched to Mac and there was a problem while exporting to (apk) 
Before anything, I have downloaded the program ( android studio) latest version. And I downloaded the required as I used to do in the Windows system.
The problem here is that when exporting, this problem occurs:
![Screenshot 2023-06-18 at 3 49 29 AM](https://github.com/airsdk/Adobe-Runtime-Support/assets/51138931/33ca4726-ef34-4835-80fc-f03fd64c7fcc)

Where, when exporting, the data is erased from the (test.xml) file without my permission, leaving only some permissions, and thus the problem occurs. I don't know why he deletes it. Unlike the Windows system, when exporting to Windows, it shows me a message about a file (test.xml ) and tells me, Do you want to make some changes? So choose No, and the export is successful, and nothing is deleted from the (test.xml) file. While on the Mac system, when exporting, it erases the data of the (test.xml) file without my permission, and thus the problem occurs. How to solve this problem so that when exporting the data is not erased from the file

**Comment by avcr2019** [comment]
any one can help ?

**Comment by marchbold** [comment]
Check which version of Java you are using and did you set the path to the Android SDK and JAVA in your adt.cfg or through the air sdk manager?

**Reply by avcr2019** [reply]
welcome my dear friend
Yes added the path in file(adt.cfg).
![Screenshot 2023-06-19 at 9 39 43 AM](https://github.com/airsdk/Adobe-Runtime-Support/assets/51138931/707ad2e0-6ad2-408a-bddc-a0b328d9e39a)

As for the version of (Java), I don't know. But I downloaded the latest Android system from the official website.
The problem here is that when the application is exported permanently, the data in file (test.xml) is erased, and therefore the application faces a problem in exporting. I don't know why the data is deleted in the file(test.xml). While in the Windows system, it does not delete it, and the export is successful. The strange thing is that the application is exported to the iPhone and does not delete the data in the (test.xml) file. It only clears it when exporting to Android.

**Comment by ajwfrost** [comment]
Not sure whether the above did solve it, but if not, can you look at the suggestions in https://github.com/airsdk/Adobe-Runtime-Support/issues/2608#issuecomment-1590720280 ? This was getting the same error "unexpected error during link"

thanks

**Reply by avcr2019** [reply]
thank you for replying
I have added this in file(xml)
`    <gradleVersion>7.5</gradleVersion>
    <androidGradlePluginVersion>7.4.0</androidGradlePluginVersion>`
And I tried and it failed again and all I did is that I added this code again in the (xml) file
When exporting again, the data in the file (xml) was erased, but I quickly made a return to the data inside the file (xml) by pressing (ctrl+z), and save and the export was successful.
Question: I tried without this code 
(    <gradleVersion>7.5</gradleVersion>
   <androidGradlePluginVersion>7.4.0</androidGradlePluginVersion>)
 and the export was successful. Should I leave it and is it useful or not. . I thank you .
Therefore, I ask you, my dear friend, do I leave this code in the file or not, and is it useful and what are its benefits
Solve this problem for those who want to solve it:
This code should be placed inside the file (xml)
`    <gradleVersion>7.5</gradleVersion>
    <androidGradlePluginVersion>7.4.0</androidGradlePluginVersion>`
When exporting, you must open a file (xml) and when deleting the data, press ((ctrl+z) and save the file during export, and the problem will be resolved

**Reply by ajwfrost** [reply]
> do I leave this code in the file or not, and is it useful and what are its benefits

If you don't need it, and the build is successful, you can omit it. The benefits may just be some additional support from Gradle that isn't actually needed, they may also have improved performance a bit. Ultimately though, if you're compiling with later target SDK and compile tools from an Android SDK, you may need to add these tags in order to get it to work..

**Reply by avcr2019** [reply]
Thank you, my dear friend, for your help.
